docs/tutorials/migrate/migrate_from_mbed_os_2_to_5.md

```diff
@@ -0,0 +1,308 @@
+### Mbed OS 2 to 5 migration guide
+
+This guide is to assist you in the process of updating an existing [component](https://os.mbed.com/components/), library or program from Mbed OS 2 to Mbed OS 5.
+
+#### Prerequisites
+
+- [Mbed CLI](https://os.mbed.com/docs/v5.7/tools/arm-mbed-cli.html)
+- An [offline compiler](/docs/latest/tools/index.html#compiler-versions)
+
+### Identifying old versions of Mbed OS
+
+We will use the Grove temperature and humidity sensor as an example: [Grove-TempHumi-Sensor](https://os.mbed.com/components/Grove-TempHumi-Sensor/)
+
+First, navigate to the Hello World repository for the component.
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/1-helloworld.png)</span>
+
+Next, view the files in the Hello World repository. The presence of an  `mbed.bld` or `mbed-rtos.lib` file signifies that this program uses an older version of Mbed OS (older than Mbed OS 5).
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/2-oldmbed.png)</span>
+
+A program that uses and has been tested with Mbed OS 5 or later has an `mbed-os.lib` file.
+
+<span class="images">![](https://os.mbed.com/media/uploads/jplunkett/3-newmbed.png)</span>
+
+Some repositories may have both an `mbed.bld` file and a `mbed-rtos.lib` file present. Mbed-RTOS is the precursor to Mbed OS 5, which combines the older Mbed OS 2 library with Mbed-RTOS. So, Mbed OS 5 can replace BOTH `mbed.bld` and `mbed-rtos.lib`.
+
+### Migrating to Mbed OS 5
+
+There are two possible outcomes when trying to migrate to Mbed OS 5:
+
+1. Replacing the old Mbed library with Mbed OS "just works," as the APIs in the library have not changed. In this instance, you're finished.
+2. Replacing the old Mbed library with Mbed OS produces some compilation errors. For one of two reasons:
+  - The API calls in the **library** are out of date, and you need to migrate to Mbed OS API syntax.
```

Comment:
The example you chose fails to compile because of target specific code in the library, so the "migrating to Mbed OS API syntax" doesn't apply to that case.

Reply:
I think this section is just talking about general cases right? Not the example yet?

Reply:
We could try to find a case of out of date API calls.

Reply:
Yes, but it makes it seem like these are the only two cases for compilation errors. There can be target specific code in application *or* library.

Reply:
How about this:

> 2. Replacing the old Mbed library with Mbed OS produces some compilation errors. These errors can occur in the library or application code for the following two reasons:
>  - The Mbed OS API calls are out of date, and you need to migrate to the updated Mbed OS API syntax.
>  - The application uses target specific code, and you need to be migrate it to use code for the specific target you are compiling for.

Reply:
> The application uses

Change to 'there is target specific code'